odl/trafos/wavelet.py

```diff
@@ -21,623 +21,242 @@
-
-    Compute a discrete 3D multiresolution wavelet reconstruction
-    from a given wavelet coefficient list.
-    Utilizes a `PyWavelet
-    <http://www.pybytes.com/pywavelets/ref/other-functions.html>`_
-    function ``pywt.dwtn``
-
-    Parameters
-    ----------
-    coeff_list : list
-        A list of wavelet approximation and detail coefficients
-        organized in the following way
-        ```[caaaN, (aadN, adaN, addN, daaN, dadN, ddaN, dddN), ...
-        (aad1, ada1, add1, daa1, dad1, dda1, ddd1)]```.
-
-        The abbreviations refer to
-
-        ``aaa`` = approx. on 1st dim, approx. on 2nd dim, approx. on 3rd dim,
-
-        ``aad`` = approx. on 1st dim, approx. on 2nd dim, detail on3rd dim,
-
-        ``ada`` = approx. on 1st dim, detail on 3nd dim, approx. on 3rd dim,
-
-        ``add`` = approx. on 1st dim, detail on 3nd dim, detail on 3rd dim,
-
-        ``daa`` = detail on 1st dim, approx. on 2nd dim, approx. on 3rd dim,
-
-        ``dad`` = detail on 1st dim, approx. on 2nd dim, detail on 3rd dim,
-
-        ``dda`` = detail on 1st dim, detail on 2nd dim, approx. on 3rd dim,
-
-        ``ddd`` = detail on 1st dim, detail on 2nd dim, detail on 3rd dim,
-
-        ``N`` = the number of scaling levels
-
-    wbasis :  ``_pywt.Wavelet``
-        Describes properties of a selected wavelet basis.
-        For more information see PyWavelet `documentation
-        <http://www.pybytes.com/pywavelets/ref/wavelets.html>`_
-
-    pad_mode : string
-        Signal extention mode. For possible extensions see the
-        `signal extenstion modes
-        <http://www.pybytes.com/pywavelets/ref/\
-signal-extension-modes.html>`_
-        of PyWavelets.
-
-    nscales : int
-        Number of scales in the coefficient list.
-
-    Returns
-    -------
-    x : `numpy.ndarray`
-        A wavalet reconstruction.
-    """
-    aaa = coeff_list[0]
-    (aad, ada, add, daa, dad, dda, ddd) = coeff_list[1]
-    coeff_dict = {'aaa': aaa, 'aad': aad, 'ada': ada, 'add': add,
-                  'daa': daa, 'dad': dad, 'dda': dda, 'ddd': ddd}
-    for tpl in coeff_list[2:]:
-        aaa = pywt.idwtn(coeff_dict, wbasis, pad_mode)
-        (aad, ada, add, daa, dad, dda, ddd) = tpl
-        coeff_dict = {'aaa': aaa, 'aad': aad, 'ada': ada, 'add': add,
-                      'daa': daa, 'dad': dad, 'dda': dda, 'ddd': ddd}
+    @property
+    def is_orthogonal(self):
+        """Whether or not the wavelet basis is orthogonal."""
+        return self.pywt_wavelet.orthogonal
 
-    x = pywt.idwtn(coeff_dict, wbasis, pad_mode)
-    return x
+    @property
+    def is_biorthogonal(self):
+        """Whether or not the wavelet basis is bi-orthogonal."""
+        return self.pywt_wavelet.biorthogonal
 
 
-class WaveletTransform(Operator):
+class WaveletTransform(WaveletTransformBase):
 
-    """Discrete wavelet trafo between discrete L2 spaces."""
+    """Discrete wavelet trafo between discretized Lp spaces."""
```

Comment:
Actually, the whole operator should be called `DiscreteWaveletTransform` similar to the situation on the Fourier side. That's because this operator is not an approximation of the continuous wavelet transform, which may also have its use cases but isn't yet implemented.